### PR TITLE
fix emit for delete on optional chain

### PR DESCRIPTION
**Pull request by ajafff**
Fixes: #35084

@rbuckton for review

### CODE REVIEW

**Comment by rbuckton** [comment]
Can you update this branch now that #34951 has been merged?

**Comment by RyanCavanaugh** [comment]
@rbuckton LGTM, care to merge?

File: src/compiler/transformers/esnext.ts [hunk 1]
```diff
@@ -152,11 +154,9 @@ namespace ts {
                 setOriginalNode(rightExpression, segment);
             }
 
-            const target = createConditional(
-                createNotNullCondition(leftExpression, capturedLeft, /*invert*/ true),
-                createVoidZero(),
-                rightExpression,
-            );
+            const target = isDelete
+                ? createConditional(createNotNullCondition(leftExpression, capturedLeft, /*negate*/ true), createTrue(), createDelete(rightExpression))
```
Comment (rbuckton) [inline src/compiler/transformers/esnext.ts]:
```suggestion
                ? createConditional(createNotNullCondition(leftExpression, capturedLeft, /*invert*/ true), createTrue(), createDelete(rightExpression))
```

File: src/compiler/transformers/esnext.ts [hunk 2]
```diff
@@ -197,5 +197,11 @@ namespace ts {
                 expression.kind !== SyntaxKind.ThisKeyword &&
                 expression.kind !== SyntaxKind.SuperKeyword;
         }
+
+        function visitDeleteExpression(node: DeleteExpression) {
+            return isOptionalChain(skipParentheses(node.expression))
+                ? setOriginalNode(visitNonOptionalExpression(node.expression, /*caputeThisArg*/ false, /*isDelete*/ true), node)
```
Comment (rbuckton) [inline src/compiler/transformers/esnext.ts]:
```suggestion
                ? setOriginalNode(visitNonOptionalExpression(node.expression, /*captureThisArg*/ false, /*isDelete*/ true), node)
```

**Comment by rbuckton** [comment]
@RyanCavanaugh: Unfortunately I can't commit the suggestions myself as part of the PR. Should I merge this as is and address separately, or wait for @ajafff to update?

**Comment by seb-saletes** [comment]
Hi, Is it release ? Do you have any kind of roadmap for bugfix like this one ?
Ty :)

**Pull request merged**